src/rest/CategoryFacetResult.ts

```diff
@@ -0,0 +1,8 @@
+import { ICategoryFacetValue } from './CategoryFacetValue';
+
+export interface ICategoryFacetResult {
+  notImplemented?: boolean;
```

Comment:
It signifies that the category facet API is not supported on the current search api. Yes it should be optionnal since it will not be returned if it is supported.

Reply:
Hmm, maybe naming the property `notImplementedBySearchApi` would give a clearer explanation of the role of the property?

Reply:
That could help, but doesn't the fact that the `notImplemented` field is in the results returned by the SearchApi already indicate that it's not implemented server side?